api/apps/api/src/modules/scenarios/input-parameter-file.provider.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { isDefined } from '@marxan/utils';
+import { ScenariosCrudService } from './scenarios-crud.service';
+
+class InputParameterFile {
+  constructor(
+    private readonly boundaryLengthModifier?: number,
+    private readonly numberOfRuns?: number,
+    private readonly marxanInputParameterFile?: Record<string, string | number>,
+  ) {}
+
+  toString(): string {
+    let builder = '';
+    if (isDefined(this.boundaryLengthModifier)) {
+      builder += `BLM ${this.boundaryLengthModifier}\n`;
+    }
+    if (isDefined(this.numberOfRuns)) {
+      builder += `NUMREPS ${this.numberOfRuns}\n`;
+    }
+    if (isDefined(this.marxanInputParameterFile)) {
+      builder += Object.entries(this.marxanInputParameterFile)
```

Comment:
AFAIR the NUMREPS and BLM are already within the `marxanInputParameterFile` as well, so it would produce this parameter twice. Not sure tho if this would break the marxan run, @aagm ?

Reply:
well spotted... IIRC from the Marxan source code, duplicated `KEY value` pairs will result in the first one to be found to be used.

If we don't want to enforce a schema for `marxanInputParameterFile` (and hence outright reject keys such as `BLM` and `NUMREPS`), then we should filter out entries with both these keys here.

Reply:
@hotzevzl what value should BLM have when `boundaryLengthModifier` is null? Should it be taken from `marxanInputParameterFile` or omitted entirely?

Reply:
@Dyostiq it is not required according to the Marxan manual and if not supplied it will default to 1.

We should not even allow `BLM` to be in `marxanInputParameterFile`: the frontend will not send it, but if a generic API clients tries to do so as part of a scenario create/update, we should reject the request.

Relatedly, we could either add known keys with default values if they are not in `marxanInputParameterFile` but I would avoid to do so - but cross checking this in Slack in a moment...

Reply:
fixed in 2c76cec1